test/conformance/channel_tracing_test.go

```diff
@@ -0,0 +1,163 @@
+// +build e2e
+
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conformance
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"knative.dev/eventing/test/base/resources"
+	"knative.dev/eventing/test/common"
+	"knative.dev/eventing/test/conformance/helpers"
+	"knative.dev/pkg/test/zipkin"
+)
+
+func TestChannelTracing(t *testing.T) {
+	testCases := map[string]struct {
+		incomingTraceId bool
+		istio           bool
+	}{
+		//"no incoming trace id": {},
+		"includes incoming trace id": {
+			incomingTraceId: true,
+		}, /*
+			"caller has istio": {
+				istio: true,
+			},*/
+	}
+
+	for n, tc := range testCases {
+		loggerPodName := "logger"
+		t.Run(n, func(t *testing.T) {
+			channelTestRunner.RunTests(t, common.FeatureBasic, func(st *testing.T, channel string) {
```

Comment:
This function needs to be defined as a helper function, if you want it to be ported to `eventing-contrib` to test other channel implementations, example https://github.com/knative/eventing/blob/master/test/conformance/helpers/channel_header_single_event_helper.go

Reply:
Done. 

I didn't know, thanks for pointing it out.